shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -997,6 +997,14 @@ public void onReceive(Context context, Intent intent) {
             this.surfaceTexture.setOnFrameAvailableListener(new SurfaceTexture.OnFrameAvailableListener() {
                 @Override
                 public void onFrameAvailable(SurfaceTexture texture) {
+                    // NB: This callback may be called on an arbitrary thread.
+                    if (SurfaceTextureRegistryEntry.this.surfaceTexture.isReleased()) { // isReleased is synchronized in JNI code.
```

Comment:
Android BUILD failed.

Unfortunately, this API only seems to be available in Android 26 and up. You will need to do a SDK version check here too :-/

Reply:
Right. I also realized that it's still racy: if all of the deinitialization happens between checking isReleased() and accessing mNativeView we'd get null anyway.

We really should pass a Handler and execute the callback on the main thread. Is nativeMarkTextureFrameAvailable actually meant to be thread-safe?